testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JDBCRealmAdminOnlyModeTestCase.java

```diff
@@ -0,0 +1,92 @@
+ */
+package integration.elytron.realm;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ADD;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.NAME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_ATTRIBUTE_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+
+import static org.junit.Assert.assertEquals;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.ModelControllerClientConfiguration;
+import org.jboss.as.test.integration.security.common.CoreUtils;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.dmr.ModelNode;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.wildfly.test.security.common.elytron.JdbcSecurityRealm;
+
+
+/**
+ * A test case to test adding a {@link JdbcSecurityRealm} within the Elytron subsystem in admin only mode.
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+@RunWith(Arquillian.class)
+public class JDBCRealmAdminOnlyModeTestCase {
+
+    protected static final int CONNECTION_TIMEOUT_IN_MS = TimeoutUtil.adjust(6 * 1000);
+    protected static final int PORT = 9990;
+    protected static final boolean ADMIN_ONLY = true;
+
+    @Test
+    public void testAddJDBCRealmAdminOnlyMode() throws Exception {
+
+        ModelControllerClient client = ModelControllerClient.Factory
+                .create(new ModelControllerClientConfiguration.Builder().setHostName(CoreUtils.getDefaultHost(false))
+                        .setPort(PORT).setConnectionTimeout(CONNECTION_TIMEOUT_IN_MS).build());
+
+        ServerReload.executeReloadAndWaitForCompletion(client, CONNECTION_TIMEOUT_IN_MS, ADMIN_ONLY,
+                TestSuiteEnvironment.getServerAddress(), PORT, null);
+
+        ModelNode operation = new ModelNode();
+        operation.get(OP_ADDR).setEmptyList();
+        operation.get(OP).set(READ_ATTRIBUTE_OPERATION);
+        operation.get(NAME).set("running-mode");
+        assertEquals("ADMIN_ONLY", client.execute(operation).get("result").asString());
+
+        operation = new ModelNode();
+        ModelNode params = new ModelNode();
+        params.get("sql").set("SELECT * FROM Users WHERE username = ?");
+        params.get("data-source").set("ExampleDS");
+
+        operation.get(OP_ADDR).add("subsystem", "elytron").add("jdbc-realm", "MyRealm");
+        operation.get("principal-query").add(params);
+        operation.get(OP).set(ADD);
```

Comment:
Again just an FYI this could be:
```
operation = Operations.createAddOperation(Operations.createAddress("subsystem", "elyrton", "jdbc-realm", "MyRealm"));
operation.get("principal-query").add(params);
```